### PR TITLE
Fix XLA symbolic shapes binding

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88928

Obsoletes https://github.com/pytorch/pytorch/pull/88772

Mostly revolves around NOT assuming that the inside is a SymNode,
but instead duck-typed to be a SymNode.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by ezyang** [comment]
cc @vanbasten23

File: torch/csrc/jit/python/init.cpp [hunk 1]
```diff
@@ -1148,38 +1148,66 @@ void initJITBindings(PyObject* module) {
...
-                                      [](c10::SymNode a) { return a->str(); })
-                                  .def("__repr__", [](c10::SymNode a) {
-                                    return a->str();
-                                  });
+      SYMNODE_UNARY(is_int)
+      SYMNODE_UNARY(is_float)
+      SYMNODE_UNARY(bool_)
+      SYMNODE_UNARY(int_)
+      SYMNODE_UNARY(sym_int)  // TODO: remove me
+      SYMNODE_UNARY(sym_float)
```
Comment (SherlockNoMad) [inline torch/csrc/jit/python/init.cpp:1164]: sym_sqrt missng?

**Review by SherlockNoMad — APPROVED** [review]

File: torch/csrc/jit/python/init.cpp (hunk 1 above)
Comment (ezyang) [reply torch/csrc/jit/python/init.cpp:1164]: in fact not missing, because this PR base is not recent enough to have sym_sqrt. I'll rebase

**Comment by ezyang** [comment]
There's also some C++ related sym_int cleanup too now.

**Comment by jinyouzhi** [comment]
Since all UT passed and no conflicts, can anyone help to merge it? thanks

**Pull request closed without merging**